src/ds/ds-options.cpp

```diff
@@ -550,16 +550,26 @@ namespace librealsense
         {
             float rv = 0.f;
             command cmd(ds::GETSUBPRESETID);
-            // if no subpreset is streaming, the firmware returns "ON_DATA_TO_RETURN" error
-            try {
-                auto res = _hwm.send(cmd);
-                // if a subpreset is streaming, checking this is the alternating emitter sub preset
-                if (res.size())
-                    rv = (res[0] == ds::ALTERNATING_EMITTER_SUBPRESET_ID) ? 1.0f : 0.f;
+            try
+            {
+                hwmon_response response;
+                auto res = _hwm.send( cmd, &response );  // avoid the throw
```

Comment:
I think we should log exceptions from `send`, or at least don't catch and dump them here, let the caller handle it.

Reply:
Send is not a good place if it's valid to return an error.
As to throwing for non-expected responses, I can make that change but have no idea what the consequences are.

Reply:
As discussed, maybe not in this PR, but after removing "expected exceptions" we should consider the right way to handle the unexpected ones.